ska_builder.py

```diff
@@ -122,8 +147,26 @@ def build_package(name):
 
     if not args.test:
         cmd_list.append("--no-test")
-    if not args.force:
+
+    if args.force:
+        for path in Path(BUILD_DIR).glob(f'*/.cache/*/{name}-*'):
+            print(f'Removing {path}')
+            path.unlink()
+
+        sys_prefix = Path(sys.prefix)
+        if (sys_prefix.parent).name == 'envs':
+            # Building in a miniconda env, can find packages one dir up in pkgs
+            pkgs_dir = sys_prefix.parent.parent / 'pkgs'
```

Comment:
I don't understand this. If I am within an environment and I run from the source skare3 directory, this will still set pkgs_dir to '/Users/javierg/SAO/miniconda3/pkgs' and delete all directories named like the package in there. Is that right? Why would conda pull use that directory? Why should this script mess with a directory out of its scope?

Reply:
This is for the corner case of building and re-building packages at the same version (and uploading and then making a new env for testing from the "official" repo). When you do that there may already be this cached version in `pkgs` and so it won't bother re-downloading it, conda just uses the existing cached version. I've been bit by this several times, along with problems related to caching in `builds`.

Reply:
so maybe adding another option like --clear-cache?

Reply:
In principle `--clear-cache` is the right thing, but in practice you need to clear the cache if and only if you are using `--force`. So I don't think the added complexity is winning us anything.

Reply:
You might not be aware of a way we (@jeanconn and I) have used this script on ocasion.

To make a quick test, you can make a change locally, commit (do not push) and build with the --force option. In this case, the current script does not realize that it is being tricked. Your change would prevent this. I think it is convenient to have a mechanism like that to test without having to push upstream.

Reply:
Of course, one could remove the output package from the output directory, reindex, and run without `--force`. But this is exactly doing by hand what `--force` is supposed to do.

Reply:
The main use case for `--force` is changing `skare3` itself (`ska_builder.py` or a recipe), since if you update the package source and commit, that makes a new version of the package (yay SCM versions!) and force is not needed anyway. So this is why I feel like there is no need for a separate cache clearing option.  I found that telling `conda build` to ignore the existing build was not enough, that I have to remove those cache files in `builds`. 

And then on the testing side if you make a new package at the same version, and then upload that new package to the web server repo, you will not get the new package with `conda create` unless you remove it from the `pkgs` cache.

Conversely, if you have not made a new package build at the same version, I can't see any reason why the cache files need to be cleared.